libs/libc/gdbstub/lib_gdbstub.c

```diff
@@ -1786,6 +1939,16 @@ FAR struct gdb_state_s *gdb_state_init(gdb_send_func_t send,
   state->priv = priv;
   state->monitor = monitor;
 
+  if (CONFIG_BOARD_MEMORY_RANGE[0] != '\0')
+    {
+      state->range = alloc_memory_region(CONFIG_BOARD_MEMORY_RANGE);
```

Comment:
could we remove all parser/allocate code and declare `CONFIG_BOARD_MEMORY_RANGE` as array format? 
`static const struct memory_region_s g_memory_regions[] = CONFIG_BOARD_MEMORY_RANGE;`
which will be beneficial for identifying bring up issues.

Reply:
Kconfig how to define a struct array？

Reply:
> Kconfig how to define a struct array？

@anjiahao1 

![image](https://github.com/user-attachments/assets/ac1c97f0-12eb-4a32-9bde-492cc102ab6f)

![image](https://github.com/user-attachments/assets/d73073e1-8c88-4ef9-ab0a-fdd3b9698c15)

FYI @xiaoxiang781216 @XuNeo

Reply:
parse code is already added before this patch, so it's better to create a new patch instead.

Reply:
All static resources could be defined in this style. Using dynamic allocator will make these debugging features unusable in some extreme scenarios.

Reply:
Yes, but it should belong to a new patch to remove ALL places.

Reply:
Great! @anchao